src/resolve/resolved-etc-hosts.c

```diff
@@ -444,5 +444,5 @@ int manager_etc_hosts_lookup(Manager *m, DnsQuestion* q, DnsAnswer **answer) {
                         return r;
         }
 
-        return 1;
+        return dns_answer_isempty(*answer) ? 0 : 1;
```

Comment:
I meant that if you do an IPv6 query (`systemd-resolve -6 heise.de`) then `manager_etc_hosts_lookup()` will return an empty response as it didn't find any AAAA record, but it will still return with 1. But that would mean that the entire lookup will fail unless we check for emptiness (i. e. to differ between NXDOMAIN and NODATA).

This still won't get us *both* the IPv4 and IPv6 address if you ask for "any family" (`systemd-resolve heise.de`), but that's within your boundaries of "if one source has any RR it should be considered sufficient".